mucommander-core/src/main/java/com/mucommander/ui/action/impl/TerminalWindow.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of muCommander, http://www.mucommander.com
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mucommander.ui.action.impl;
+
+import com.jediterm.pty.PtyProcessTtyConnector;
+import com.jediterm.terminal.TtyConnector;
+import com.jediterm.terminal.ui.JediTermWidget;
+import com.jediterm.terminal.ui.UIUtil;
+import com.jediterm.terminal.ui.settings.DefaultSettingsProvider;
+import com.pty4j.PtyProcess;
+import com.pty4j.PtyProcessBuilder;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Based on JediTerm's BasicTerminalShellExample class.
+ */
+public final class TerminalWindow {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(TerminalWindow.class);
+
+    private TerminalWindow() {
+    }
+
+    public static JediTermWidget createTerminal(String currentFolder) {
+        return createTerminalWidget(currentFolder);
+    }
+
+    private static JediTermWidget createTerminalWidget(String currentFolder) {
+        JediTermWidget widget = new JediTermWidget(80, 24, new DefaultSettingsProvider());
+        widget.setTtyConnector(createTtyConnector(currentFolder));
+        widget.start();
+        return widget;
+    }
+
+    private static @NotNull TtyConnector createTtyConnector(String currentFolder) {
+        try {
+            Map<String, String> envs = System.getenv();
+            String[] command;
+            if (UIUtil.isWindows) {
```

Comment:
we already have a per-OS module, how about moving this part to those (mucommander-os-*) modules?

Reply:
hmm, I will take a look and comment later.